core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1043,13 +1045,33 @@ class KafkaApis(val requestChannel: RequestChannel,
         offsetFetchResponse
       } else {
         // Clients are not allowed to see offsets for topics that are not authorized for Describe.
-        val (authorizedOffsets, _) = authHelper.partitionSeqByAuthorized(
+        val authorizedNames = authHelper.filterByAuthorized(
           requestContext,
           DESCRIBE,
           TOPIC,
           offsetFetchResponse.topics.asScala
         )(_.name)
-        offsetFetchResponse.setTopics(authorizedOffsets.asJava)
+
+        val topics = new mutable.ArrayBuffer[OffsetFetchResponseData.OffsetFetchResponseTopics]
+        offsetFetchResponse.topics.forEach { topic =>
+          if (authorizedNames.contains(topic.name)) {
+            if (useTopicIds) {
+              // If the topic is not provided by the group coordinator, we set it
+              // using the metadata cache.
+              if (topic.topicId == Uuid.ZERO_UUID) {
+                topic.setTopicId(metadataCache.getTopicId(topic.name))
+              }
+              // If we don't have the topic id at all, we skip the topic because
+              // we can not serialize it without it.
```

Comment:
IMHO, we could put this comment at the end of the block.
In this way, this comment will be more specific to the condition without the topic ID.

Reply:
Hum... I am not sure to follow your suggestion. I usually put the comment related to a if statement right before it or at the beginning of the block. Putting it as the end of the block is weird.

Reply:
I mislook.
It will still be `ZERO_UUID` if the corresponding topic ID doesn't exist in the metadata cache.
Sorry for the confusing comment.